hms-app/src/components/RelevantIdeasLoader.tsx

```diff
@@ -39,15 +39,15 @@ export default function RelevantIdeasLoader({
     requiredSkills: [],
     category: undefined,
     creationDate: new Date(),
-  } as Idea)
+  })
 
-  const loadSelectedHackathonData = () => {
-    getHackathonDetails(selectedHackweek).then((data) => {
+  const loadSelectedHackathon = () => {
+    getHackathonDetails(selectedHackathonID).then((data) => {
       setHackathonData({
         id: data.id,
         title: data.title,
-        startDate: data.startDate,
-        endDate: data.endDate,
+        startDate: new Date(data.startDate),
+        endDate: new Date(data.endDate),
         participants: data.participants,
         categories: data.categories,
         ideas: data.ideas,
```

Comment:
You can make `getHackatonDetails()` return a `Promise<Hackathon>` and then you can have here simply:
```
  const hackathonMap = getHackatonsForContext()
    .map(hackathon => mapHackatonToSelectItem(hackathon))
```
So you don't need to copy one object state to other by hand